rskj-core/src/test/java/co/rsk/peg/federation/FederationStorageProviderImplTests.java

```diff
@@ -10,152 +9,104 @@
+            Arguments.of(P2SH_ERP_FEDERATION_FORMAT_VERSION, null),
+            Arguments.of(INVALID_FEDERATION_FORMAT, null),
+            Arguments.of(EMPTY_FEDERATION_FORMAT, null),
+            Arguments.of(INVALID_FEDERATION_FORMAT, createFederation(STANDARD_MULTISIG_FEDERATION_FORMAT_VERSION)),
+            Arguments.of(EMPTY_FEDERATION_FORMAT, createFederation(STANDARD_MULTISIG_FEDERATION_FORMAT_VERSION))
         );
     }
 
-    private void testGetNewFederation(
+    @ParameterizedTest
+    @MethodSource("provideFederationAndFormatArguments")
+    void testGetNewFederation(
         int federationFormat,
-        Federation storedFederation
+        Federation expectedFederation
     ) {
         // Arrange
-        Repository repository = mock(Repository.class);
-
-        // Mock federation format in storage
-        when(repository.getStorageBytes(
-            PrecompiledContracts.BRIDGE_ADDR,
-            NEW_FEDERATION_FORMAT_VERSION.getKey())
-        ).thenReturn(BridgeSerializationUtils.serializeInteger(federationFormat));
 
-        // Mock federation
-        byte[] serializedFederation = null;
-        if (storedFederation != null) {
-            serializedFederation = BridgeSerializationUtils.serializeFederation(storedFederation);
-        }
-
-        when(repository.getStorageBytes(
-            PrecompiledContracts.BRIDGE_ADDR,
-            NEW_FEDERATION_KEY.getKey())
-        ).thenReturn(serializedFederation);
-
-        FederationStorageProvider federationStorageProvider = createFederationStorageProvider(repository);
+        StorageAccessor storageAccessor = new InMemoryStorage();
+        byte[] federationFormatSerialized = getFederationFormatSerialized(federationFormat);
+        storageAccessor.saveToRepository(NEW_FEDERATION_FORMAT_VERSION.getKey(), federationFormatSerialized);
+        byte[] serializedFederation = getSerializedFederation(expectedFederation, federationFormat);
+        storageAccessor.saveToRepository(NEW_FEDERATION_KEY.getKey(), serializedFederation);
+        FederationStorageProvider federationStorageProvider = new FederationStorageProviderImpl(storageAccessor);
 
         // Act
-        Federation obtainedFederation = federationStorageProvider.getNewFederation(federationConstantsRegtest, activations);
 
-        // Assert
+        Federation obtainedFederation = federationStorageProvider.getNewFederation(federationConstants, activations);
 
-        // Assert that the NEW_FEDERATION_FORMAT_VERSION key is read from the storage
-        verify(repository, times(1)).getStorageBytes(
-            PrecompiledContracts.BRIDGE_ADDR,
-            NEW_FEDERATION_FORMAT_VERSION.getKey()
-        );
+        // Directly saving a null federation in storage to then assert that the method returns the cached federation
+        if(nonNull(expectedFederation)) {
+            storageAccessor.saveToRepository(NEW_FEDERATION_KEY.getKey(), null);
+        }
```

Comment:
im not a fan of doing this kind of things. I suggest having one specific test to check for cache